README Updates for 1.0.0 (#523)
aws-amplify/amplify-android: README.md

<img src="https://s3.amazonaws.com/aws-mobile-hub-images/aws-amplify-logo.png" alt="AWS Amplify" width="550">

 <a href="https://discord.gg/jWVbPfC" target="_blank">
   <img src="https://img.shields.io/discord/308323056592486420?logo=discord"" alt="Discord Chat" />  
 </a>

AWS Amplify provides a high-level interface to perform different categories of
cloud operations. Each category is fulfilled by a _plugin_. You specify which
plugins to use during setup.

The default plugins that we provide are designed to facilitate interaction with
Amazon Web Services (AWS). But, the Amplify framework is designed to be
extensible to any other backend or service.

To familiarize yourself with Amplify, checkout our [Getting Started
Guide](https://docs.amplify.aws/lib/getting-started/setup/q/platform/android).

## Features / APIs

> - **[Analytics](https://docs.amplify.aws/lib/analytics/getting-started/q/platform/android)**

Reviewer: Should we sort these in "Marketing-order", instead of lexicographical order?

Author: Alphabetical was a punt. What's that order?

  Easily collect analytics data for your app. Analytics data includes user

>   sessions and other custom events that you want to track in your app.

Reviewer: Duplicative of previous sentence, perhaps just -- 
```suggestion
  sessions and other custom events.
```

- **[API
  (GraphQL)](https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/android)**

>   Interact with your GraphQL server or AWS AppSync API with an easy-to-use &
>   configured GraphQL client.

Reviewer: We should disband from unquantified phrases "simple" or "easy-to-use." To Amazon-ify such statements, we could corroborate the in terms of # of steps to get started -- or # of lines of code to get started, etc. The user can then decide for themselves whether or not it is easy (hopefully.)

```suggestion
  Designed for interaction with your GraphQL server or AWS AppSync API
```

Author: I copied all of this copy from the JavaScript README, with the thought that if we want to optimize we can do it across all three.

Reviewer: Alright, cool. I don't love their descriptions, but there is a certain value in keeping the messaging consistent, I suppose.

- **[API
  (REST)](https://docs.amplify.aws/lib/restapi/getting-started/q/platform/android)**

>   Provides a simple solution when making HTTP requests. It provides an
>   automatic, lightweight signing process which complies with AWS Signature
>   Version 4.

Reviewer: Same deal - simple, lightweight, automatic ...
```suggestion
  Communicate with a preconfigured REST endpoint. Automatically signs requests using AWS Signature Version 4.
```

Author: Same as above. Stolen from JS, I think, with the idea that we should optimize across and be consistent in our messaging.

- **[Authentication](https://docs.amplify.aws/lib/auth/getting-started/q/platform/android)**

>   APIs and building blocks for developers who want to create user authentication
>   experiences.

Reviewer: We have to be really careful with `API` in the docs, cause the eponymous category.

```suggestion
  Uniquely identify your app's users, and define their access to different resources.
```

- **[DataStore](https://docs.amplify.aws/lib/datastore/getting-started/q/platform/android)**

>   Provides a programming model for leveraging shared and distributed data

Reviewer: Is it a programming model?

Author: I nicked this from the What's New announcement post: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-amplify-datastore/

}
```

> ### AWS Mobile Client

Reviewer: Should we instead include a note about setting up Auth, before API and DataStore will work? It could still be a worthwhile call out. The use of a disparate library to do auth was a surprise in preview. The need to use Auth is less of a surprise, but still not immediately obvious to a new comer, I'd suppose.

Author: Yes, I think that's a good idea! I'll write something up.

Author: Wrote something tiny.
